monkey/monkey_island/cc/repository/mongo_credentials_repository.py

```diff
@@ -68,7 +68,7 @@ def _get_credentials_from_collection(self, collection) -> Sequence[Credentials]:
     def _save_credentials_to_collection(self, credentials: Sequence[Credentials], collection):
         try:
             for c in credentials:
-                encrypted_credentials = self._encrypt_credentials_mapping(Credentials.to_mapping(c))
+                encrypted_credentials = self._encrypt_credentials_mapping(c.dict())
```

Comment:
Unit tests should fail. If they're not, there's a bug in the tests or `.dict()` is ok. I'm _pretty_ sure `.dict()` should break tests, so  it's probably a bug in the tests. We should investigate.

Reply:
Seems to work just fine without `simplify`

```
Python 3.10.4 (main, Jun 29 2022, 12:14:53) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from common.credentials import Credentials, Password
>>> p = Password(password='pw')
>>> c = Credentials(identity=None, secret=p)
>>> print(c.dict())
{'identity': None, 'secret': {'password': 'pw'}}
>>> 
```

I'll spend some time investigating when you do and don't need `simplify`. Ithink it's that `pydantic` always handles `pydantic` models, as long as all fields are serializable, it's fine. But if you use a `IPv4Address` or something, that's when you need `simplify`. I'll confirm.

Reply:
Confirmed:
```
from pathlib import Path
from ipaddress import IPv4Address
from common.credentials import Credentials, Password
from common.base_models import InfectionMonkeyBaseModel

p = Password(password='pw')
c = Credentials(identity=None, secret=p)

print(c.dict())

class TestModel(InfectionMonkeyBaseModel):
    credentials: Credentials
    ip: IPv4Address
    home: Path


t = TestModel(credentials=c, ip=IPv4Address("127.0.0.1"), home=Path("/home/user"))

print(t.dict())

```

```
$ python3 test.py
{'identity': None, 'secret': {'password': 'pw'}}
{'credentials': {'identity': None, 'secret': {'password': 'pw'}}, 'ip': IPv4Address('127.0.0.1'), 'home': PosixPath('/home/user')}

```

Conclusion: There's nothing wrong with the unit tests. Credentials work without `simplify` (for now). However, it's probably safer to just use `simplify` to avoid issues in the future.